modules/hostap/src/supp_api.c

```diff
@@ -309,6 +311,14 @@ static int wpas_add_and_config_network(struct wpa_supplicant *wpa_s,
 		_wpa_cli_cmd_v("set_network %d mode 2", resp.network_id);
 	}
 
+	strncpy(ssid_null_terminated, params->ssid, WIFI_SSID_MAX_LEN);
+	ssid_null_terminated[params->ssid_length] = '\0';
+	params->ssid = ssid_null_terminated;
```

Comment:
is this necessary? can't we just use it locally? Ideally the `params` should have been `const`.
Also, if you are updating `params->ssid` then you also need to update `params->ssid_length`. Same for `psk_length` too.

Reply:
We are passing only null terminated ssid/psk from shell. I think ssid_length/psk_length should be ssid/psk without '\0' only. If someone use hostap and pass ssid/psk without null terminator this will add by default.